app.rb

```diff
@@ -26,15 +28,18 @@ def add_person()
     when 1
       puts 'Has parent permission? [Y/N]'
       parent_permission = Input.new.parent_permission
-      student = Student.new(age, parent_permission, name)
+      type = 'Student'
+      student = Student.new(type, parent_permission, name, age)
+      @persons = []
       @persons.push(student)
```

Comment:
Whenever I  added a person of type `Student`, instead of being added alongside what was already in the list, it overrites the previously added student. In the end, I end up with a list holding only the last student added. Kindly fix this issue. Line 33 might have something to do with it.